### PR TITLE
Switch Site: remove site icons from All Sites button

### PR DESCRIPTION
**Pull request by vindl**
#### Proposed Changes

Remove the site icons from the All Sites button to reduce visual clutter. I don't think these added much value, so I experimented with removing them to make the site selector less noisy. This also more closely replicates the design mock-ups that @SaxonF shared for the project.

|Before|After|
|-|-|
|Screenshot 2022-10-07 at 18 40 54|Screenshot 2022-10-07 at 18 40 24|

| Before hover | After hover |
|-|-|
|Screenshot 2022-10-07 at 18 44 21|Screenshot 2022-10-07 at 18 44 41|

| Jetpack Cloud before | Jetpack Cloud After |
|-|-|
|Screenshot 2022-10-07 at 19 21 03|Screenshot 2022-10-07 at 19 20 41|
|Screenshot 2022-10-07 at 19 21 39|Screenshot 2022-10-07 at 19 22 28|

#### Testing Instructions

1. Use a test account that has more than one site.
2. Click on Stats (or Domains, Plugins, Posts, Pages).
3. Click on Switch Site.
4. Verify that the icons no longer appear.
5. Test in supported browsers.
6. Ensure that there are no other usages of the deleted component in the codebase.
7. Ensure that there are no other usages of `All Sites` where `sites` prop is passed.
8. Test in Calypso Green (Jetpack Cloud) locally with `CALYPSO_ENV=jetpack-cloud-development yarn run start`.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

**Review by zaguiini — COMMENTED** [review]
Thanks for the PR! Some thoughts on the changes:

1. It makes sense to remove the `sites` prop on the usages, as well as the mentions in the README file;
2. We still need to accept a `count` prop there, otherwise it would probably break the behavior of https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-site-network/index.jsx#L49.

Furthermore, the behavior we currently have on `trunk` is already mismatching what we want. For example, I have `111` sites in my account, but the `<AllSites />` element reports only `73`.

That number -- `73` -- represents the number of visible sites, but we're not having this distinction anymore in Calypso, so we need to adapt the site count logic in the Redux connector:

```diff
diff --git a/client/blocks/all-sites/index.jsx b/client/blocks/all-sites/index.jsx
index acea602b40..1e21f3cda6 100644
--- a/client/blocks/all-sites/index.jsx
+++ b/client/blocks/all-sites/index.jsx
@@ -5,7 +5,6 @@ import PropTypes from 'prop-types';
 import { Component } from 'react';
 import { connect } from 'react-redux';
 import Count from 'calypso/components/count';
-import { getCurrentUserVisibleSiteCount } from 'calypso/state/current-user/selectors';
 import getSites from 'calypso/state/selectors/get-sites';
 
 import './style.scss';
@@ -78,11 +77,13 @@ class AllSites extends Component {
 // don't instantiate function in `connect`
 const isSiteVisible = ( { visible = true } ) => visible;
 
-export default connect( ( state ) => {
-	const visibleSites = getSites( state )?.filter( isSiteVisible );
-	const userSitesCount = getCurrentUserVisibleSiteCount( state );
+export default connect( ( state, props ) => {
+	const sites = getSites( state, { shouldSort: false } ).filter( Boolean );
+	const userSitesCount = props.count ?? sites.length;
 
 	return {
-		count: config.isEnabled( 'realtime-site-count' ) ? visibleSites.length : userSitesCount,
+		count: config.isEnabled( 'realtime-site-count' )
+			? sites.filter( isSiteVisible ).length
+			: userSitesCount,
 	};
 } )( localize( AllSites ) );

```

This diff modifies the props in a way that gives:
- the `count` value if it's provided as a prop;
- the visible sites count in case we're browsing Jetpack Cloud;
- the total sites (visible + hidden) count to match the behavior on the Site Selector.

Hope it helps, and let me know if there are any questions!

**Comment by vindl** [comment]
> We still need to accept a count prop there, otherwise it would probably break the behavior of https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-site-network/index.jsx#L49.

Good catch! I audited the `sites` prop usages but forgot to do the same for `count`.

File: client/blocks/all-sites/README.md [hunk 1]
```diff
@@ -14,7 +14,6 @@ function render() {
 
 ## Props
 
-- `sites (array)` - An array of `site` objects.
```
Comment (vykes-mac) [inline client/blocks/all-sites/README.md:17]: should we also remove the `sites` param in 
```
function render() {
	return <AllSites sites={ sitesArray } />;
}
```

**Review by vykes-mac — COMMENTED** [review]
It tests well based on the conditions described in the testing instructions. There doesn't seem to be any reference to `AllSites` with sites prop being passed expect the example in `client/blocks/all-sites/README.md`

File: client/blocks/all-sites/README.md (hunk 1 above)
Comment (vindl) [reply client/blocks/all-sites/README.md:17]: Yes. 👍

**Comment by danielbachhuber** [comment]
Looks so much better! 🔥

**Comment by mpkelly** [comment]
Looks neater than before. I tested using Calypso Live and Chrome.

File: client/blocks/all-sites/README.md (hunk 1 above)
Comment (sejas) [reply client/blocks/all-sites/README.md:17]: @vykes-mac @vindl I've committed this change here https://github.com/Automattic/wp-calypso/pull/68785/commits/3d40b3213e87bcb8026153568f800ea02a7e960a

**Review by sejas — APPROVED** [review]
I like this change. Looking good to me.

About the sites counting difference is something that already occurs in production, not introduced by this PR. We could create a new Issue/PR to conciliate that.

**Comment by zaguiini** [comment]
>About the sites counting difference is something that already occurs in production, not introduced by this PR. We could create a new Issue/PR to conciliate that.

That's fair, @sejas! Would you mind creating a follow-up PR, or a new issue, so it can be worked on later this week?

**Comment by sejas** [comment]
> That's fair, @sejas! Would you mind creating a follow-up PR, or a new issue, so it can be worked on later this week?

- @zaguiini  Created the follow-up issue https://github.com/Automattic/wp-calypso/issues/68919 and started a discussion there about that change.

**Comment by vindl** [comment]
> We still need to accept a count prop there, otherwise it would probably break the behavior of https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-site-network/index.jsx#L49.

@zaguiini I brought back the count prop override in f83de4b5c3c480936987afb312d2d3a77c26470b to preserve the existing behavior in `plugin-site-network`.

I decided not to change other aspects of counting logic since it's out of scope of this PR, and it's not what we want atm based on the discussion in this issue: https://github.com/Automattic/wp-calypso/issues/68919.

**Review by zaguiini — APPROVED** [review]

**Pull request merged**